xmodule/capa_block.py

```diff
@@ -1433,6 +1444,8 @@ def closed(self):
         """
         Is the student still allowed to submit answers?
         """
+        if self.course_is_archived():
+            return True
```

Comment:
Would it be more direct to just check course.has_ended() here? I'm wary of adding a new method just for this. I'm not sure how well understood "archived" is within the codebase.